libraries/node-core-library/src/FileSystem.ts

```diff
@@ -829,25 +890,80 @@ export class FileSystem {
    * @param linkPath - The path to the link.
    */
   public static getRealPath(linkPath: string): string {
-    return fsx.realpathSync(linkPath);
+    return FileSystem._wrapException(() => {
+      return fsx.realpathSync(linkPath);
+    });
   }
 
   /**
    * An async version of {@link FileSystem.getRealPath}.
    */
   public static getRealPathAsync(linkPath: string): Promise<string> {
-    return fsx.realpath(linkPath);
+    return FileSystem._wrapExceptionAsync(() => {
+      return fsx.realpath(linkPath);
+    });
   }
 
   // ===============
   // UTILITY FUNCTIONS
   // ===============
 
   /**
-   * Returns true if the error provided indicates the file or folder
-   * does not exist.
+   * Returns true if the error provided indicates the file or folder does not exist.
    */
-  public static isNotExistError(error: NodeJS.ErrnoException): boolean {
-    return error.code === 'ENOENT' || error.code === 'ENOTDIR';
+  public static isNotExistError(error: Error): boolean {
+    return FileSystem.isFileDoesNotExistError(error) || FileSystem.isFolderDoesNotExistError(error);
+  }
+
+  /**
+   * Returns true if the error provided indicates the file does not exist.
+   */
+  public static isFileDoesNotExistError(error: Error): boolean {
+    return FileSystem._isErrnoExceptionType(error) && (error.code === 'ENOENT');
+  }
+
+  /**
+   * Returns true if the error provided indicates the folder does not exist.
+   */
+  public static isFolderDoesNotExistError(error: Error): boolean {
+    return FileSystem._isErrnoExceptionType(error) && (error.code === 'ENOTDIR');
+  }
+
+  private static _isErrnoExceptionType(error: Error): error is NodeJS.ErrnoException {
```

Comment:
This would actually be pretty useful as a public API.  Suggest to shorten the name to `isErrnoException()`.

Reply:
```ts
/**
 * Returns true if the object is a system error that implements the 
 * `NodeJS.ErrnoException` interface.
 */
```